### PR TITLE
Version tag updates

### PR DESCRIPTION
**Pull request by SmittieC**
## Description

- All AI responses are now tagged with which version generated it, not only those generated by a version
- Pipeline generate AI messages are now also tagged like experiment generated AI messages
- The experiment session table now shows all versions that the session spans, not just the latest one.
- Working version generated AI responses are tagged with an "unreleased" tag (to be confirmed/changed when we sort out the nomenclature), indicating that this is not a version. One "issue" with this is that if we create a new version, then chatting to a working version will still tag responses with the "unreleased" tag even though there are no changes between the current working version and the latest version.

Options:
- Store an "is_dirty" type of flag on the experiments model. Whenever we update any related model (source material, pipelines etc), we must diff the updated object with its previous version. If there are changes, we update all experiments using this objects to be "dirty"
- Do a diff inside the state where we need to figure out if the "unreleased" tag needs to be set or not. Con: On each user query we'll do the expensive diff check. Pro: This is only when chatting to a working version. Public chats will not cause the expensive DB queries to run.
- Any other?

#### Session table shows versions that the session spans
image

#### We indicate that the chat happened on an unreleased version
image

### Docs

N/A

### CODE REVIEW

**Comment by SmittieC** [comment]
cc @bderenzi

**Comment by SmittieC** [comment]
@coderabbitai do a full review

File: apps/service_providers/llm_service/state.py [hunk 1]
```diff
@@ -178,9 +178,13 @@ def save_message_to_history(self, message: str, type_: ChatMessageType, experime
 
         if type_ == ChatMessageType.AI:
             self.ai_message = chat_message
-            if not self.experiment.is_working_version:
+            chat_message.add_system_tag(
+                tag=f"v{self.experiment.version_number}", tag_category=TagCategories.EXPERIMENT_VERSION
+            )
```
[the remark this answers is not in the document]
Comment (SmittieC) [reply apps/service_providers/llm_service/state.py:183]: It's always available

File: apps/service_providers/llm_service/state.py [hunk 2]
```diff
@@ -178,9 +178,13 @@ def save_message_to_history(self, message: str, type_: ChatMessageType, experime
...
+            if self.experiment.is_working_version:
                 chat_message.add_system_tag(
-                    tag=self.experiment.version_display, tag_category=TagCategories.EXPERIMENT_VERSION
+                    tag="unreleased",
+                    tag_category=TagCategories.VERSION_DEPLOYED_STATUS,
```
[the remark this answers is not in the document]
Comment (SmittieC) [reply apps/service_providers/llm_service/state.py:187]: ChatExperimentState is different to ExperimentAssistantState. This need no update, but thanks

File: apps/pipelines/models.py [hunk 3]
```diff
@@ -181,12 +182,22 @@ def _create_pipeline_run(self, input: PipelineState, session: ExperimentSession)
...
-        return ChatMessage.objects.create(
+        chat_message = ChatMessage.objects.create(
             chat=session.chat,
             message_type=type_.value,
             content=message,
         )
-        # TODO: Add tags here?
+
+        if type_ == ChatMessageType.AI:
+            self.ai_message = chat_message
```
Comment (snopoke) [inline apps/pipelines/models.py:192]: I don't think we need to save the ai message to the node

**Review by snopoke — COMMENTED** [review]
Generally seems OK though I wonder about using a separate tag for the 'unreleased' vs changing the name of the version tag e.g. `v3 unreleased`.

**Comment by snopoke** [comment]
> * One "issue" with this is that if we create a new version, then chatting to a working version will still tag responses with the "unreleased" tag even though there are no changes between the current working version and the latest version.

I don't think this is worth trying to work around.

**Comment by SmittieC** [comment]
> Generally seems OK though I wonder about using a separate tag for the 'unreleased' vs changing the name of the version tag e.g. `v3 unreleased`.

I like this suggestion. I'm going with it

**Review by snopoke — APPROVED** [review]

**Pull request merged**